### DISCUSSION TITLE
TP 7 - Sobre los "archivos de práctica anteriores" y la realización de los ejercicios

### DISCUSSION THREAD

**Discussion by gascatalan3** [INGCOM-UNRN-P1/cursada-2023#190] [Preguntas y respuestas] [answered]
Buenas!
Algunas consultas, porque no quedó muy claro... entonces, ¿cada ejercicio del TP7 los hacemos en un DIRECTORIO aparte cada uno y al git subimos el directorio?

Otra consulta, respecto a los archivos de prácticas anteriores, mediante el #include "practica3.c", ¿debo tener en ese directorio, de cada ejercicio individual, la práctica o los ejercicios que voy a usar pero que NO contengan el main? 

Y una última consulta, el usar ejercicios de TP's anteriores ¿aplica solo para el ejercicio 2, 3 y el 4 opcional cierto? Porque el ejercicio 1 yo lo hice sin ningun ejercicio de TP's anteriores y entiendo que es así...

Gracias!

**Comment by Nikzza** [comment]
Con respecto a esa pregunta se refiere a hacer funciones en módulos diferentes y subir tanto las funciones como el main

![image](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/141664052/67466aab-2c32-4b40-9389-6613e6a7f086)


> Otra consulta, respecto a los archivos de prácticas anteriores, mediante el #include "practica3.c", ¿debo tener en ese directorio, de cada ejercicio individual, la práctica o los ejercicios que voy a usar pero que NO contengan el main?

a lo que se refiere es que tenes que incluir como librería los módulos donde están tus funciones, ellas tienen que estar sin el main

>Y una última consulta, el usar ejercicios de TP's anteriores ¿aplica solo para el ejercicio 2, 3 y el 4 opcional cierto? Porque el ejercicio 1 yo lo hice sin ningun ejercicio de TP's anteriores y entiendo que es así...

exacto, no todos los ejercicios se basan en esos tps tenes que ir armandolos vos tambien

**Comment by martinvilu** [comment, marked as answer]
>  Algunas consultas, porque no quedó muy claro... entonces, ¿cada ejercicio del TP7 los hacemos en un DIRECTORIO aparte cada uno y al git subimos el directorio?

Sí

Por ejemplo
```
ejercicio1/cadenas.c
ejercicio1/main.c
ejercicio2/arreglos.c
ejercicio2/main.c
```
La compilación es por medio de donde esté el `main`, es la razón por la cual, lo que hagas `#include` no puede tener `main` o nombres de funciones repetidos.

**Reply by gascatalan3** [reply]
Okey. Una duda más, entonces yo lo que compilo es el archivo main, y el archivo de arreglos.c y cadenas.c, que me pusiste en este ejemplo, son los archivos que yo "incluyo" con el #include en el main.c?

**Reply by Nikzza** [reply]
si
un ejemplo #include "arreglos.c"

**Reply by gascatalan3** [reply]
Ok. Voy a probar y ver si lo puedo hacer. Muchas gracias!
